dts/arm/renesas/gen3/rcar_gen3_cr7.dtsi

```diff
@@ -70,6 +71,76 @@
 			reg = <0xe6060000 0x50c>;
 		};
 
+		pwm0: pwm@e6e30000 {
+			compatible = "renesas,rcar-pwm";
+			reg = <0xe6e30000 0x8>;
+			#pwm-cells = <2>;
+			clocks = <&cpg CPG_MOD 523>,
+				<&cpg CPG_CORE CPG_CORE_CLK_S0D12>;
+			status = "disabled";
+			label = "pwm0";
+		};
+
+		pwm1: pwm@e6e31000 {
+			compatible = "renesas,rcar-pwm";
+			reg = <0xe6e31000 0x8>;
+			#pwm-cells = <2>;
+			clocks = <&cpg CPG_MOD 523>,
+				<&cpg CPG_CORE CPG_CORE_CLK_S0D12>;
+			status = "disabled";
+			label = "pwm1";
+		};
+
+		pwm2: pwm@e6e32000 {
+			compatible = "renesas,rcar-pwm";
+			reg = <0xe6e32000 0x8>;
+			#pwm-cells = <2>;
+			clocks = <&cpg CPG_MOD 523>,
+				<&cpg CPG_CORE CPG_CORE_CLK_S0D12>;
+			status = "disabled";
+			label = "pwm2";
+		};
+
+		pwm3: pwm@e6e33000 {
+			compatible = "renesas,rcar-pwm";
+			reg = <0xe6e33000 0x8>;
+			#pwm-cells = <2>;
+			clocks = <&cpg CPG_MOD 523>,
+				<&cpg CPG_CORE CPG_CORE_CLK_S0D12>;
+			status = "disabled";
+			label = "pwm3";
+		};
+
+		pwm4: pwm@e6e34000 {
```

Comment:
"This LSI incorporates seven channels for R-Car H3, R-Car H3-N, R-Car M3-W, R-Car M3-N, R-Car E3; five channels
for R-Car V3M and R-Car V3H; four channels for R-Car D3 pulse width modulation (PWM) timer."

Since all Gen3 SoCs have different number of pwm channels we may move these pwm definition in the SoC specific dts "r8a77951.dtsi".